task/mgmt-gateway/src/update/host_flash.rs

```diff
@@ -0,0 +1,315 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::{common::CurrentUpdate, ComponentUpdater};
+use crate::mgs_handler::{BufferMutex, UpdateBuffer};
+use core::ops::Range;
+use drv_gimlet_hf_api::{
+    HfDevSelect, HfError, HfMuxState, HostFlash, PAGE_SIZE_BYTES,
+    SECTOR_SIZE_BYTES,
+};
+use gateway_messages::{
+    ComponentUpdatePrepare, ResponseError, SpComponent, UpdateId,
+    UpdateInProgressStatus, UpdatePreparationProgress, UpdatePreparationStatus,
+    UpdateStatus,
+};
+
+userlib::task_slot!(HOST_FLASH, hf);
+
+pub(crate) struct HostFlashUpdate {
+    task: HostFlash,
+    current: Option<CurrentUpdate<State>>,
+}
+
+impl HostFlashUpdate {
+    pub(crate) fn new() -> Self {
+        Self {
+            task: HostFlash::from(HOST_FLASH.get_task_id()),
+            current: None,
+        }
+    }
+}
+
+impl ComponentUpdater for HostFlashUpdate {
+    fn block_size(&self) -> usize {
```

Comment:
This doesn't use `self`, so could be a `const fn block_size() -> usize`

(which would then make it easier to use in an explicit `const` calculation of buffer size)

Reply:
Traits can't have `const fn`s yet, but I changed it to an associated `const` instead in f4743daf.

I think originally I was hoping to stack an array of `&'static dyn ComponentUpdater` objects, in which casing having it as a `&self` function would still be useful. But for now we only have one implementer of this, so I went ahead with the change (plus I can use these constants when determining `UPDATE_BUFFER_SIZE`, which is handy). Can always revisit if we end up wanting trait objects at some point.